src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -3259,4 +3283,43 @@ mod tests {
             assert_eq!(got, exp);
         }
     }
+
+    #[test]
+    fn test_instr_binary() {
+        let cases: Vec<(&str, &str, i64)> = vec![
+            ("a", "abcdefg", 1),
+            ("0", "abcdefg", 0),
+            ("c", "abcdefg", 3),
+            ("F", "abcdefg", 0),
+            ("cd", "abcdefg", 3),
+            (" ", "abcdefg", 0),
+            ("", "", 0),
```

Comment:
the result is  `1` in mysql.

Reply:
@AndreMouche Thanks for the review.The same problem in #4337 .